### PR TITLE
[MDS-5505] Cypress test major projects Part 1

### PR DESCRIPTION
**Pull request by henryoforeh-dev**
## Objective 

[MDS-5505](https://bcmines.atlassian.net/browse/MDS-5505)

- Implemented cypress test to upload and download documents in mine major projects.
- Refactored JwtManager in document manager to allow for cypress authentication and verification. Note:* I used exactly the same pattern used in core-api to achieve this.
- Ran regression test to ensure document related functionality is not broken.
- I originally implemented the major project's Cypress test, but I made the decision to temporarily remove it. I'm waiting for the pull request (PR) to be merged before re-implementing the test. Once the PR is successfully merged, and the deployment to Docman is completed, I will reintroduce the test in part 2 of the PR.
- Added keycloak as part of the containers that should spin up in docker.

### CODE REVIEW

File: cypress/keycloak-users.json [hunk 1]
```diff
@@ -1,84 +1,77 @@
...
-        "realmRoles": ["admin"]
-      }
-    ]
-  }
-]
+[ {
+  "realm" : "standard",
+  "users" : [ {
+    "id" : "af681d51-6743-4fca-9256-f5321cbc0aa7",
+    "createdTimestamp" : 1694039547332,
```
Comment (taraepp) [inline cypress/keycloak-users.json]: what has actually changed here?

**Review by matbusby-fw — DISMISSED** [review]
I'm not super familiar with the jwt_manager.py setup, but from what I can tell it looks good!

File: services/document-manager/backend/app/utils/access_decorators.py [hunk 2]
```diff
@@ -60,7 +67,12 @@ def wrapper(*args, **kwds):
 def _inner_wrapper(func, role):
     @wraps(func)
     def wrapper(*args, **kwds):
-        return jwt.requires_roles([role])(func)(*args, **kwds)
+            try:
+                return getJwtManager().requires_roles([role])(func)(*args, **kwds)
+            except AuthError as e:
+                print("required: ", role)
+                print("found: ", getJwtManager().get_all_roles())
+                raise Forbidden(e.error['description'])
```
Comment (taraepp) [inline services/document-manager/backend/app/utils/access_decorators.py:75]: I like this

File: docker-compose.M1.yaml [hunk 3]
```diff
@@ -101,6 +100,7 @@ services:
       - redis
       - jaeger
       - otelcollector
+      - keycloak
```
Comment (taraepp) [inline docker-compose.M1.yaml:121]: I think this file also needs the definition added for keycloak like it is in the other file

Comment (henryoforeh-dev) [reply docker-compose.M1.yaml:121]: Missed it, thanks for the catching it.

**Review by taraepp — DISMISSED** [review]

**Review by isuru-aot — APPROVED** [review]

**Review by taraepp — APPROVED** [review]

**Review by simensma-fresh — APPROVED** [review]

**Pull request merged**